internal/pkg/skuba/upgrade/addon/version.go

```diff
@@ -45,6 +45,19 @@ func UpdatedAddons(clusterVersion *version.Version) (AddonVersionInfoUpdate, err
 
 	for addonName, version := range latestAddonVersions {
 		skubaConfigVersion := skubaConfig.AddonsVersion[addonName]
+		if skubaConfigVersion == nil {
+			skubaConfigVersion = &kubernetes.AddonVersion{
+				// Assign current version as updated version and manifest-version to 0 if skuba-config not exist since:
+				//
+				// 1) For better user experience on `skuba addon upgrade plan` if this addon does not updated
+				//    `skuba addon upgrade plan` won't show need upgrade for this addon.
+				//
+				// 2) If addon image version bumped, developer needs to bump manifest version also.
```

Comment:
This should not be needed if using appropriate schema for tags, see https://github.com/SUSE/avant-garde/issues/905. Am I missing something? If not I would not include this comment here.

Reply:
No, this PR is for fixing for the customers who had installed CaaSP v4 with skuba `1.0.2` already.

For the next coming release, we will release a new skuba from the master branch with the tag `X.Y.Z`. The customers updated his/her `skuba` with version `X.Y.Z` and he/she wanna use new skuba command to upgrade cluster or upgrade addon. Then, the customer will bump into this issue by running `skuba addon upgrade [plan|apply]`.

Reply:
But this won't happen if we use the `x.y.z-revk` tags. I am not concerned about the code logic, just about the comment `2) If addon image version bumped, ...`. I believe it is not needed to update the manifest if the image version bumps. It is just a matter of using the correct tag, isn't it?

Reply:
Yes, take a look at Rafa's [comment](https://github.com/SUSE/skuba/pull/726#pullrequestreview-293964654)

Reply:
Ok I see, then I am fine, thanks for the feedback and details :+1: